chore(project): dep and build tasks update to aurelia rc1

doktordirk opened the pull request:
includes
 https://github.com/SpoonX/aurelia-orm/pull/145
https://github.com/SpoonX/aurelia-orm/pull/153

doktordirk committed:
feat(entity): allow custom id property names. Use @idProperty() decorator to set a custom id property name

doktordirk committed:
fix(entity): use getId for collections

doktordirk committed:
doc(project): docBlocks

doktordirk committed:
doc(decorators): add association examples

doktordirk committed:
doc(decorator): add info about assocication vs model naming

doktordirk committed:
chore(project): dep and build tasks update to aurelia rc1

doktordirk committed:
chore(build): added missing build dep gulp-empty

doktordirk committed:
chore(project): set eslint consistent-return:0

RWOverdijk commented on .eslintrc.json:
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "./node_modules/aurelia-tools/.eslintrc.json",
+  "rules": {
+    "no-new-func": 0,
```
I don't agree with these two exceptions personally. Especially consistent returns are important.

doktordirk committed:
refactor(entiity): de-nest if

doktordirk committed:
chore(project): depdendencies fix

doktordirk committed:
feat(entity): prevent accidental infinite loops

fixed tests. server needs to return an id on saving.  since it'll make it too complicated to also return fake associated entity ids, the test got simplified there

doktordirk committed:
chore(logger): revert moving to own file

doktordirk committed:
Merge branch 'refactor-endpoint-warning' into dep-update

doktordirk committed:
Merge branch 'feat-setData-markClean' into dep-update

doktordirk committed:
Merge branch 'feat-get-clean-entity' into dep-update

doktordirk committed:
Merge branch 'feat-custom-id-property' into dep-update

# Conflicts:
#	doc/api_entity.md

doktordirk committed:
Merge branch 'doc-docBlocks' into dep-update

# Conflicts:
#	src/entity.js

doktordirk committed:
fix(entity): fix reset and test

doktordirk committed:
Merge branch 'upstream' into dep-update

doktordirk committed:
chore(project): bring back 'consistent return' rule
